### ISSUE TITLE
Add ComparisonTester to guava-testlib

### ISSUE THREAD

**Issue by PhilippWendler** [google/guava#2223]
A `ComparisonTester` that is similar to `EqualsTester` but tests a `compareTo` method or a `Comparator` would be great to have.

Possible interface:
```
public class ComparableTester {
   public ComparableTester addEquivalenceGroup(Object... equivalenceGroup);
   public ComparableTester testCompareTo();
   public ComparableTester testComparator(Comparator<?> comparator);
}
```
The expected order of objects could be simply defined by the order of calls to `addEquivalenceGroup`.

The class would test that objects within each equivalence group are deemed equivalent, that the order of equivalence groups matches the actual ordering, that `compareTo` is reflexive etc.

Testing `compareTo` of `Comparable` classes could then also be added to `ClassSanityTester` and `AbstractPackageSanityTests`.

**Comment by cpovirk** [comment]
Yep. We have one of these internally (with a slightly different but broadly similar API -- we should consider switching to yours), and we've been meaning to release. But obviously we haven't done that :)

**Comment by cgruber** [comment]
Just had one of our Oracle colleagues ask if we had something like this.  I guess a lot of folks who don't use ComparisonChain end up implementing contract-violating compareTo methods, and they were hoping to recommend something like guava-testlib to help them.

**Comment by PhilippWendler** [comment]
I recently went over all `Comparator`s and all `compareTo` implementations in our code base, and indeed the majority of those that did not use ComparisonChain were wrong. So open-sourcing your ComparisonTester would be really appreciated.

**Comment by PhilippWendler** [comment]
Are there still chances of getting this open-sourced?

**Comment by nickita-khylkouski** [comment]
Hi @cpovirk,

I'd like to help move this forward. There are two options:

1. **If you can share your internal API/implementation**, I can help adapt it for open-source release, matching Guava's style and adding tests.

2. **If that's not possible**, I can implement it from scratch based on the proposed API in this issue. The implementation would:
   - Test reflexivity: `compare(x, x) == 0`
   - Test symmetry: `sgn(compare(x, y)) == -sgn(compare(y, x))`
   - Test transitivity: `compare(x, y) > 0 && compare(y, z) > 0` implies `compare(x, z) > 0`
   - Test equivalence groups are actually equivalent
   - Test ordering matches the declaration order
   - Handle `Comparable.compareTo()` and `Comparator.compare()` uniformly

Which approach would you prefer? Happy to contribute either way.
